internal/backend/builder.go

```diff
@@ -5,22 +5,616 @@ package backend
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
 
+	containerd "github.com/containerd/containerd/v2/client"
+	"github.com/containerd/containerd/v2/core/images"
+	"github.com/containerd/containerd/v2/core/images/archive"
+	"github.com/containerd/errdefs"
+	"github.com/containerd/log"
 	"github.com/containerd/nerdctl/v2/pkg/api/types"
 	"github.com/containerd/nerdctl/v2/pkg/buildkitutil"
-	"github.com/containerd/nerdctl/v2/pkg/cmd/builder"
+	"github.com/containerd/nerdctl/v2/pkg/clientutil"
+	"github.com/containerd/nerdctl/v2/pkg/containerutil"
+	"github.com/containerd/nerdctl/v2/pkg/platformutil"
+	"github.com/containerd/nerdctl/v2/pkg/referenceutil"
+	"github.com/containerd/nerdctl/v2/pkg/strutil"
+	"github.com/containerd/platforms"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
 //go:generate mockgen --destination=../../mocks/mocks_backend/nerdctlbuildersvc.go -package=mocks_backend github.com/runfinch/finch-daemon/internal/backend NerdctlBuilderSvc
 type NerdctlBuilderSvc interface {
-	Build(ctx context.Context, client ContainerdClient, options types.BuilderBuildOptions) error
+	Build(ctx context.Context, client ContainerdClient, options types.BuilderBuildOptions, buildID string) error
 	GetBuildkitHost() (string, error)
 }
 
-func (*NerdctlWrapper) Build(ctx context.Context, client ContainerdClient, options types.BuilderBuildOptions) error {
-	return builder.Build(ctx, client.GetClient(), options)
+func (w *NerdctlWrapper) Build(ctx context.Context, client ContainerdClient, options types.BuilderBuildOptions, buildID string) error {
+	return w.RunBuild(ctx, client.GetClient(), options, buildID)
 }
 
 func (w *NerdctlWrapper) GetBuildkitHost() (string, error) {
 	return buildkitutil.GetBuildkitHost(w.globalOptions.Namespace)
 }
+
+type PlatformParser interface {
+	Parse(platform string) (platforms.Platform, error)
+	DefaultSpec() platforms.Platform
+}
+
+type platformParser struct{}
+
+func (p platformParser) Parse(platform string) (platforms.Platform, error) {
+	return platforms.Parse(platform)
+}
+
+func (p platformParser) DefaultSpec() platforms.Platform {
+	return platforms.DefaultSpec()
+}
+
+// TODO: Make generateBuildctlArgs public in nerdctl library.
+//
+//nolint:stylecheck // adding a todo for exposing env in nerdctl library
+func (w *NerdctlWrapper) RunBuild(ctx context.Context, client *containerd.Client, options types.BuilderBuildOptions, buildID string) error {
```

Comment:
Can we make it clear that this code comes from nerdctl's `pkg/cmd/builder/build.go`?